### ISSUE TITLE
Only disk available is `/boot/efi`

### ISSUE THREAD

**Issue by shawnyeager** [fflewddur/tophat#141]
First, thanks for this very useful extension. I appreciate it.

On Ubuntu 24.04.1, using ZFS with encryption, Tophat only recognizes `/boot/efi`. No drop down is shown. See screenshot.

![Screenshot from 2024-09-20 22-23-31](https://github.com/user-attachments/assets/dbe2cfe9-4501-4e3d-a302-715916c1059f)

**Comment by fflewddur** [comment]
This should be fixed as of TopHat v18. Can you confirm whether this is still an issue?

**Comment by shawnyeager** [comment]
Thanks, @fflewddur. Given recent design changes, where only activity , not free capacity, is shown for disk, this wouldn't be apparent, no?

**Comment by shawnyeager** [comment]
Correction: v17 is still what's available via Gnome Extensions.

**Comment by fflewddur** [comment]
You're right, v18 is still being reviewed for extensions.gnome.org. For now you can manually download and install it from https://github.com/fflewddur/tophat/releases/tag/v18. If you edit the metadata.json file and decrease the version to 17, then GNOME won't try to "upgrade" back to v17 on each logout.

**Comment by shawnyeager** [comment]
Thanks, @fflewddur. I did install manually, then logged out and back in. Unfortunately, it's only seeing my `/boot` mount. See attached.

![2024-12-31_13-45](https://github.com/user-attachments/assets/648ff6bb-cceb-4a57-ac9a-a1ffdf6ed369)

![2024-12-31_13-45_1](https://github.com/user-attachments/assets/9aa842e1-452f-4d16-b0c0-d71234c94038)

![2024-12-31_13-45_2](https://github.com/user-attachments/assets/e0483d5b-ad77-4675-9a9f-3b0a658c2757)

**Comment by fflewddur** [comment]
*sigh*

Could you share the output of the `df -P` command with me? TopHat uses that to identify disks, and I'm guessing there's something I'm not parsing correctly for encrypted ZFS partitions.

**Comment by shawnyeager** [comment]
Appreciate your efforts. FYI that mine is LUKS-encrypted btrfs, not ZFS. Here you go:

```
 /usr/bin/df -P
Filesystem       1024-blocks      Used Available Capacity Mounted on
dev                 16043800         0  16043800       0% /dev
run                 16074420      2428  16071992       1% /run
efivarfs                 148        70        74      49% /sys/firmware/efi/efivars
/dev/mapper/root   975695576 146830060 828555028      16% /
tmpfs               16074420         0  16074420       0% /dev/shm
tmpfs                   1024         0      1024       0% /run/credentials/systemd-journald.service
tmpfs               16074424     10908  16063516       1% /tmp
/dev/mapper/root   975695576 146830060 828555028      16% /.snapshots
/dev/mapper/root   975695576 146830060 828555028      16% /var/cache/pacman/pkg
/dev/mapper/root   975695576 146830060 828555028      16% /var/log
/dev/mapper/root   975695576 146830060 828555028      16% /home
/dev/nvme0n1p1       1046512    230908    815604      23% /boot
tmpfs                3214884       252   3214632       1% /run/user/1000
```

**Comment by fflewddur** [comment]
Ah, I think I found the issue. If you change the regex for RE_DF_IS_DISK in ~/.local/share/gnome-shell/extensions/tophat@fflewddur.github.io/helpers.js from `/^\s*\/dev\/(\w+)(.*)$/` to `/^\s*\/dev\/(\S+)(.*)$/`, that should allow it to find your / partition.

**Comment by shawnyeager** [comment]
Can confirm that works. 👍🏼

Still not sure that `Automatic` works, but selecting the partition does. Thanks.

**Comment by shawnyeager** [comment]
I've also installed the update on a second machine (Intel vs AMD on my primary, if that matters). This second machine is also Arch, Gnome 47, LUKS-encrypted btrfs. I've just noticed that "Top Processes" doesn't populate.

![image](https://github.com/user-attachments/assets/5ca51664-ea98-4679-aaa3-fa16036b7bd7)

**Comment by fflewddur** [comment]
Sorry for the issue with the top processes not working; this is resolved as of today's release.

**Comment by shawnyeager** [comment]
Works perfectly. Thanks, @fflewddur.
